compiler/rustc_middle/src/ty/context.rs

```diff
@@ -2895,9 +2895,7 @@ impl<'tcx> TyCtxt<'tcx> {
         self.mk_bound_variable_kinds(
             self.late_bound_vars_map(id.owner)
                 .and_then(|map| map.get(&id.local_id).cloned())
-                .unwrap_or_else(|| {
-                    bug!("No bound vars found for {:?} ({:?})", self.hir().node_to_string(id), id)
-                })
+                .unwrap_or_default()
```

Comment:
Because we're now walking up the whole def-id tree from the MIR body we're checking up to its typeck root, collecting all of the late-bound items, we are now calling `late_bound_vars` on intermediate items that don't have late-bound vars collected. 

I guess I could also check the def kind before calling `late_bound_vars`.

Reply:
Replaced with a check on the def_kind for is_fn_like